src/controllers/scripting/legacy/controllerscriptinterfacelegacy.cpp

```diff
@@ -756,81 +820,107 @@ bool ControllerScriptInterfaceLegacy::isScratching(int deck) {
         pScratch2Enable->set(activate ? 1 : 0);
     }
 
-    // used in scratchProcess for the different timer behavior we need
-    m_brakeActive[deck] = activate;
-    double initRate = rate;
-
-    if (activate) {
-        // store the new values for this spinback/brake effect
-        if (initRate == 1.0) { // then rate is really 1.0 or was set to default
-            // in /res/common-controller-scripts.js so check for real value,
-            // taking pitch into account
-            initRate = getDeckRate(group);
+    // Used for killing the current timer when both enabling or disabling
+    // Don't kill timer yet! This may be a brake init while currently spinning back
+    // and we don't want to interrupt that.
+    int timerId = m_scratchTimers.key(deck);
+
+    if (!activate) {
+        m_brakeActive[deck] = false;
+        m_spinbackActive[deck] = false;
+        stopScratchTimer(timerId);
+        return;
+    }
+    // Distinguish spinback and brake. Both ramp to a very low rate to avoid a long,
+    // inaudible run out. For spinback that rate is negative so we don't cross 0.
+    // Spinback and brake also require different handling in scratchProcess.
+    double initRate;
+    if (rate < -kBrakeRampToRate) { // spinback
+        m_spinbackActive[deck] = true;
+        m_brakeActive[deck] = false;
+        m_rampTo[deck] = -kBrakeRampToRate;
+        initRate = rate;
+    } else if (rate > kBrakeRampToRate) { // brake
+        // It just doesn't make sense to allow brake to interrupt spinback or an
+        // already running brake process
+        if (m_spinbackActive[deck] || m_brakeActive[deck]) {
+            return;
         }
-        // stop ramping at a rate which doesn't produce any audible output anymore
-        m_rampTo[deck] = 0.01;
-        // if we are currently softStart()ing, stop it
+        m_brakeActive[deck] = true;
+        m_spinbackActive[deck] = false;
+        m_rampTo[deck] = kBrakeRampToRate;
+        // Let's fetch the current rate to create a seamless brake process
+        initRate = getDeckRate(group);
+        // If we are currently softStart'ing adopt the current scratch rate
         if (m_softStartActive[deck]) {
             m_softStartActive[deck] = false;
             AlphaBetaFilter* filter = m_scratchFilters[deck];
             if (filter != nullptr) {
                 initRate = filter->predictedVelocity();
             }
         }
+    } else { // -kBrakeRampToRate <= rate <= kBrakeRampToRate
+        // This filters stopped deck and rare case of very low initial rates
+        m_brakeActive[deck] = false;
+        m_spinbackActive[deck] = false;
+        stopScratchTimer(timerId);
+        stopDeck(group);
+        return;
+    }
+    stopScratchTimer(timerId);
 
-        // setup timer and set scratch2
-        timerId = startTimer(kScratchTimerMs);
-        m_scratchTimers[timerId] = deck;
+    // If we want to brake or spin back and are currently softStart'ing, stop it.
```

Comment:
I just forgot to remove this line in #4708 
After this PR is merged I'll remove it, push to 2.3, merge that to main